addon/components/slack-search-input/component.js

```diff
@@ -236,7 +236,12 @@ export default Ember.Component.extend({
           this.attrs.enter();
         }
       } else if (keyCode === KEYS.ESC) {
-        this._mainInput.blur();
+        e.preventDefault();
+        if (this.attrs.escape) {
+          this.attrs.escape();
```

Comment:
maybe also make if `escape` returns `true` do default behaviour that is `this._mainInput.blur();` ?